### ISSUE TITLE
unused_crate_dependencies false positives for dev-dependencies added for Doctests

### ISSUE THREAD

**Issue by laralove143** [rust-lang/rust#101334]
I tried this code:

```toml
[dev-dependencies]
once_cell = "1.14"
```
```rust
/// ```rust
/// use once_cell::sync::Lazy;
/// ```
fn foo() {}
```

I expected to see this happen: *No warning since once_cell is used in the Doctest*

Instead, this happened: *The warning is still emitted*

### Meta

`rustc --version --verbose`:
```
rustc 1.62.1 (e092d0b6b 2022-07-16)
binary: rustc
commit-hash: e092d0b6b43f2de967af0887873151bb1c0b18d3
commit-date: 2022-07-16
host: aarch64-apple-darwin
release: 1.62.1
LLVM version: 14.0.5
```

**Comment by epage** [comment]
As part of https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#lintscargo, Cargo has a lint that wraps `unused_crate_dependencies`, see https://doc.rust-lang.org/nightly/cargo/reference/lints.html#unused_dependencies.

Instead of reporting false positives for doctests, Cargo will ignore dev-dependencies.  I've added logging so you can see the raw information but it is pretty messy because we don't keep as much information for log-only reporting.
